### PR TITLE
statistics: support extend show histograms

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/36481

Problem Summary:

### What is changed and how it works?

support extend show histogrames to show memory status

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: build/nogo_config.json [hunk 1]
```diff
@@ -244,7 +244,7 @@
       ".*_generated\\.go$": "ignore generated code"
     },
     "only_files": {
-      "planner/core/rule_partition_eliminate.go": "planner/core/rule_partition_eliminate code",
```
Comment (hawkingrei) [inline build/nogo_config.json:247]: Why? You only need to add new item.

Comment (Yisaer) [reply build/nogo_config.json:247]: we don't have `rule_partition_eliminate.go`

File: executor/show_stats.go [hunk 2]
```diff
@@ -176,18 +176,20 @@ func (e *ShowExec) appendTableForStatsHistograms(dbName, tblName, partitionName
 		return
 	}
 	for _, col := range stableColsStats(statsTbl.Columns) {
-		// Pass a nil StatementContext to avoid column stats being marked as needed.
-		if col.IsInvalid(nil, false) {
+		if !col.IsStatsInitialized() || col.TotalRowCount() == 0 {
```
Comment (xuyifangreeneyes) [inline executor/show_stats.go:179]: When `col.TotalRowCount()` is zero, can we also display it? So we can distinguish between uninitialized stats and zero-count stats.

**Review by xuyifangreeneyes — COMMENTED** [review]
rest LGTM

**Review by xuyifangreeneyes — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**